core/lib/src/response/status.rs

```diff
@@ -275,6 +363,70 @@ impl<'r, R: Responder<'r>> Responder<'r> for NotFound<R> {
     }
 }
 
+
+/// Sets the status of the response to 409 (Conflict).
+///
+/// If a responder is supplied, the remainder of the response is delegated to
+/// it. If there is no responder, the body of the response will be empty.
+///
+#[derive(Debug, Clone, PartialEq)]
+pub struct Conflict<R>(pub Option<R>);
+
+/// Sets the status code of the response to 409 Conflict. If the responder is
+/// `Some`, it is used to finalize the response.
+impl<'r, R: Responder<'r>> Responder<'r> for Conflict<R> {
+    fn respond_to(self, req: &Request<'_>) -> Result<Response<'r>, Status> {
+        let mut build = Response::build();
+        if let Some(responder) = self.0 {
+            build.merge(responder.respond_to(req)?);
+        }
+
+        build.status(Status::Conflict).ok()
+    }
+}
+
+/// Sets the status of the response to 410 (Gone).
+///
+/// If a responder is supplied, the remainder of the response is delegated to
+/// it. If there is no responder, the body of the response will be empty.
+///
+#[derive(Debug, Clone, PartialEq)]
+pub struct Gone<R>(pub Option<R>);
+
+/// Sets the status code of the response to 410 Gone. If the responder is
+/// `Some`, it is used to finalize the response.
+impl<'r, R: Responder<'r>> Responder<'r> for Gone<R> {
+    fn respond_to(self, req: &Request<'_>) -> Result<Response<'r>, Status> {
+        let mut build = Response::build();
+        if let Some(responder) = self.0 {
+            build.merge(responder.respond_to(req)?);
+        }
+
+        build.status(Status::Gone).ok()
+    }
+}
+
+/// Sets the status of the response to 451 (Unavailable For Legal Reasons).
+///
+/// If a responder is supplied, the remainder of the response is delegated to
+/// it. If there is no responder, the body of the response will be empty.
+///
+#[derive(Debug, Clone, PartialEq)]
+pub struct UnavailableForLegalReasons<R>(pub Option<R>);
```

Comment:
`UnavailableForLegalReasons(R)` is such a specific error code I don't expect this one to be used much; my guess is that this status code is more likely to appear in in custom-implemented or derived `Responder`s. Of course `rocket::response::status::Custom(Status::UnavailableForLegalReasons, error)` is always an option too.